phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionScannerFactory.java

```diff
@@ -367,10 +367,6 @@ private boolean scanTillScanStartRow(final RegionScanner s,
           if (isDummy(result)) {
             return true;
           }
-          if (arrayFuncRefs != null && arrayFuncRefs.length > 0 && arrayKVRefs.size() > 0) {
-            int arrayElementCellPosition = replaceArrayIndexElement(arrayKVRefs, arrayFuncRefs, result);
-            arrayElementCell = result.get(arrayElementCellPosition);
```

Comment:
This never worked. Someone did not understand how Java references work. This just changed the local arrayElemenCell reference with no effect outside of the method.

Reply:
So true, much needed change 👍